src/components/geocoder-panel.js

```diff
@@ -133,97 +83,95 @@ function isValid(key) {
+      mapboxApiAccessToken: PropTypes.string.isRequired,
+      mapState: PropTypes.object.isRequired,
+      updateVisData: PropTypes.func.isRequired,
+      removeDataset: PropTypes.func.isRequired,
+      updateMap: PropTypes.func.isRequired,
+
+      transitionDuration: PropTypes.number,
+      width: PropTypes.number
+    };
+
+    removeGeocoderDataset() {
+      this.props.removeDataset(GEOCODER_DATASET_NAME);
+    }
 
-  onSelected = (viewport = null, geoItem) => {
-    const {
-      center: [lon, lat],
-      text,
-      bbox
-    } = geoItem;
-    this.removeGeocoderDataset();
-    this.props.dispatch(
-      addDataToMap({
-        datasets: [generateGeocoderDataset(lat, lon, text)],
-        options: {
+    onSelected = (viewport = null, geoItem) => {
+      const {
+        center: [lon, lat],
+        text,
+        bbox
+      } = geoItem;
+      this.removeGeocoderDataset();
+      this.props.updateVisData(
+        [generateGeocoderDataset(lat, lon, text)],
+        {
           keepExistingConfig: true
         },
-        config: {
+        KeplerGlSchema.parseSavedConfig({
           version: 'v1',
           config: {
             visState: {
               layers: [ICON_LAYER]
             }
           }
-        }
-      })
-    );
-    this.props.dispatch(
-      fitBounds(bbox || [lon - GEO_OFFSET, lat - GEO_OFFSET, lon + GEO_OFFSET, lat + GEO_OFFSET])
-    );
-    this.setState({
-      selectedGeoItem: geoItem
-    });
-  };
+        })
+      );
+      const bounds = bbox || [
+        lon - GEO_OFFSET,
+        lat - GEO_OFFSET,
+        lon + GEO_OFFSET,
+        lat + GEO_OFFSET
+      ];
+      const {center, zoom} = geoViewport.viewport(bounds, [
+        this.props.mapState.width,
+        this.props.mapState.height
+      ]);
+
+      this.props.updateMap({
+        latitude: center[1],
+        longitude: center[0],
+        zoom,
+        pitch: 0,
+        bearing: 0,
+        transitionDuration: this.props.transitionDuration,
+        transitionInterpolator: new FlyToInterpolator()
+      });
+    };
+
+    removeMarker = () => {
+      this.removeGeocoderDataset();
+    };
+
+    render() {
+      const {isGeocoderEnabled, mapboxApiAccessToken, width} = this.props;
+      return (
+        <StyledGeocoderPanel
+          className="geocoder-panel"
+          width={width}
+          style={{display: isGeocoderEnabled ? 'block' : 'none'}}
+        >
+          {isValid(mapboxApiAccessToken) && (
+            <Geocoder
+              mapboxApiAccessToken={mapboxApiAccessToken}
+              onSelected={this.onSelected}
+              onDeleteMarker={this.removeMarker}
+              width={width}
+            />
+          )}
+        </StyledGeocoderPanel>
+      );
+    }
+  }
 
-  removeMarker = () => {
-    this.setState({
-      selectedGeoItem: null
-    });
-    this.removeGeocoderDataset();
+  GeocoderPanel.defaultProps = {
+    transitionDuration: 3000
```

Comment:
Isn't it too long?

Reply:
Probably should be calculated in future update. For example if I'm zoomed to Texas and first I look for Dallas an then Austin it would be annoying to wait 3 seconds for such transition.